tools/SourceKit/lib/SwiftLang/SwiftDocSupport.cpp

```diff
@@ -1467,8 +1474,7 @@ SourceFile *SwiftLangSupport::getSyntacticSourceFile(
       InputFile(InputBuf->getBufferIdentifier(), /*isPrimary*/false, InputBuf,
                 file_types::TY_Swift));
 
-  if (ParseCI.setup(Invocation)) {
-    Error = "Compiler invocation set up failed";
+  if (ParseCI.setup(Invocation, Error)) {
```

Comment:
Just noticed this case appears to only need the compiler instance for syntactic queries, but uses `initCompilerInvocationNoInputs` which will set the requested action to `Typecheck` and therefore I *believe* will now load the stdlib where it didn't previously. Should we overwrite the requested action before calling `setup`, or perhaps plumb through a parameter on `initCompilerInvocationNoInputs` to determine whether we want a semantic action?

Reply:
Ah, good catch. I added a commit to explicitly specify the frontend action when setting up the compiler invocation. This surfaced another case where we don’t need to load the stdlib either.